src/org/omegat/core/team2/impl/GITCredentialsProvider.java

```diff
@@ -307,45 +323,71 @@ public boolean supports(CredentialItem... items) {
      * @return true when entered, false on cancel.
      */
     private Credentials askCredentials(URIish uri, Credentials credentials) {
-        GITUserPassDialog userPassDialog = new GITUserPassDialog(Core.getMainWindow().getApplicationFrame());
-        userPassDialog.setLocationRelativeTo(Core.getMainWindow().getApplicationFrame());
-        userPassDialog.descriptionTextArea.setText(OStrings
-                .getString(credentials.username == null ? "TEAM_USERPASS_FIRST" : "TEAM_USERPASS_WRONG", uri.getHumanishName()));
-        // if username is already available in uri, then we will not be asked for an username, so we cannot
-        // change it.
-        if (uri.getUser() != null && !"".equals(uri.getUser())) {
-            userPassDialog.userText.setText(uri.getUser());
-            userPassDialog.userText.setEditable(false);
-            userPassDialog.userText.setEnabled(false);
-        }
-        if (credentials.username != null) {
-            userPassDialog.userText.setText(credentials.username);
-        }
-        userPassDialog.setVisible(true);
-        if (userPassDialog.getReturnStatus() == GITUserPassDialog.RET_OK) {
-            credentials.username = userPassDialog.userText.getText();
-            credentials.password = new String(userPassDialog.passwordField.getPassword());
-            return credentials;
+        IMainWindow mw = Core.getMainWindow();
+        if (mw != null) {  // when GUI omegat application
```

Comment:
I would split the branches of this `if` into separate methods: `askCredentialsGui` and `askCredentialsCli`